src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
Do you think it still makes sense to do an early exit if the value of the `key` amounts to a char array of 0s and thus will result in a noop for the xor?